src/auslib/services/releases.py

```diff
@@ -491,3 +568,125 @@ def set_release(name, blob, product, old_data_versions, when, changed_by, trans)
+
+        if not dbo.hasPermission(changed_by, "release", "delete", product, trans):
+            raise PermissionDeniedError(f"{changed_by} is not allowed to delete {product} releases")
+
+        dbo.releases_json.scheduled_changes.delete(where={"base_name": name}, old_data_version=old_data_version, changed_by=changed_by, transaction=trans)
+
+        for asset in dbo.release_assets.scheduled_changes.select(
+            where={"base_name": name}, columns=[dbo.release_assets.scheduled_changes.base_path, dbo.release_assets.scheduled_changes.data_version]
+        ):
+            dbo.release_assets.scheduled_changes.delete(
+                where={"base_name": name, "base_path": asset["base_path"]}, old_data_version=asset["data_version"], changed_by=changed_by, transaction=trans
+            )
+
+
+def set_read_only(name, read_only, old_data_version, changed_by, trans):
+    product = dbo.releases_json.select(where={"name": name}, columns=[dbo.releases_json.product], transaction=trans)[0]["product"]
+
+    # If the Release is being changed to read-write, it may require signoff
+    use_sc = False
+    if read_only is False:
+        live_on_product_channels = []
+        stmt = select([dbo.rules.rule_id, dbo.rules.product, dbo.rules.channel]).where(
+            ((dbo.releases_json.name == dbo.rules.mapping) | (dbo.releases_json.name == dbo.rules.fallbackMapping)) & (dbo.releases_json.name == name)
+        )
+        for row in trans.execute(stmt).fetchall():
+            live_on_product_channels.append(dict(row))
+
+        if live_on_product_channels:
+            log.debug(f"{name} is live on {live_on_product_channels}")
+            prs = dbo.rules.getPotentialRequiredSignoffs(live_on_product_channels, transaction=trans)
+            # If the Release is mapped to by a Rule that requires signoff, we cannot proceed
+            if any([v for v in prs.values()]):
+                use_sc = True
+
+            # If it wasn't mapped to by a Rule that requires signoff, check for _any_ required signoffs
+            # for its product. This is a bit aggressive, but it protects against Releases for important
+            # products (eg: Firefox) from being modified before they go live on a protected channel.
+            if dbo.productRequiredSignoffs.select(where={"product": product}, transaction=trans):
+                use_sc = True
+
+    permission = "unset" if read_only else "set"
+    if not dbo.hasPermission(changed_by, "release_read_only", permission, product, trans):
+        raise PermissionDeniedError(f"{changed_by} is not allow to {permission} read_only for {product} releeases")
+
+    if use_sc:
+        data = dbo.releases_json.select(where={"name": name}, columns=[dbo.releases_json.data], transaction=trans)[0]["data"]
+        sc_id = dbo.releases_json.scheduled_changes.insert(
+            name=name,
+            product=product,
+            data=data,
+            data_version=old_data_version,
```

Comment:
`read_only` is missing from the list of arguments but you have tests like `test_set_readwrite_scheduled_change_because_of_rule` to cover this case. Does this work because the column default is false, and a scheduled change may only be required to go read/write (ie false) ? A comment may be in order.

Reply:
You're right about why this works, but it's still an oversight. Better to be explicit here.